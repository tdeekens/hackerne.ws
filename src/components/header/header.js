import React from 'react';
import styles from './header.module.css';

const isUserOnline = () =>
  typeof window !== 'undefined' && window.navigator && window.navigator.onLine;

export const Header = props => (
  <div className={styles.header}>
    <span>
      Your latest hackerne.ws top stories{' '}
      {!isUserOnline() ? " (you're offline)" : null}
    </span>
  </div>
);
Header.displayName = 'Header';

export default Header;
