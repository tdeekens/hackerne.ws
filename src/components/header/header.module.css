.header {
   background-color: #f60;
   position: fixed;
   z-index: 999;
   height: 55px;
   position: sticky;
   margin: 0;
   padding: 0;

   display: flex;
   align-items: center;
}

.header > span {
   margin: 0 auto;
   color: white;
}
