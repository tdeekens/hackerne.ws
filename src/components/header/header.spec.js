import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './header';

beforeAll(() => {
  global.window = {};
});
describe('rendering', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
