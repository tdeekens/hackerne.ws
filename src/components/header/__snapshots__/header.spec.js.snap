// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<div>
  <span>
    Your latest hackerne.ws top stories
     
     (you're offline)
  </span>
</div>
`;
